lib/lz4.h

```diff
@@ -179,13 +179,21 @@ LZ4LIB_API int LZ4_compress_fast (const char* src, char* dst, int srcSize, int d
 
 
 /*!
+LZ4_compress_fast_extState_noReset() :
 LZ4_compress_fast_extState() :
     Same compression function, just using an externally allocated memory space to store compression state.
     Use LZ4_sizeofState() to know how much memory must be allocated,
     and allocate it on 8-bytes boundaries (using malloc() typically).
     Then, provide it as 'void* state' to compression function.
+
+    Use the _noReset variant if LZ4_resetStream() was called on the state
+    buffer before being used for the first time (calls to both extState
+    functions leave the state in a safe state, so zeroing is not required
+    between calls). Otherwise, using the legacy _extState requires LZ4 to
+    reinitialize the state internally for every call.
 */
 LZ4LIB_API int LZ4_sizeofState(void);
+LZ4LIB_API int LZ4_compress_fast_extState_noReset (void* state, const char* src, char* dst, int srcSize, int dstCapacity, int acceleration);
```

Comment:
This is a new prototype. In theory, we are now bound to support it forever (at least, until the major library version changes).

I was wondering : would it be better to first declare it within a "staging area", so that we get some time to use it and make sure it fulfill its role correctly ? (Something equivalent to `ZSTD_STATIC_LINKING_ONLY` ?)

A recent example where such staging area proved useful is `LZ4F_compressFrame_usingCDict()`, which prototype is going to change now as your experiments proved it would be better to add a `CCtx` as argument.

Reply:
Yes, good call. It needs to be accessible to lz4frame.c, and it would be nice to make it accessible externally, since it does provide better performance. But I agree it should be marked unstable.

Reply:
We have the [end of `lz4.h`](https://github.com/lz4/lz4/blob/dev/lib/lz4.h#L344) used for such purpose. There is no protection macro defined though, for historical reasons (programs which already use those definitions should not break now). But we could introduce one for newer candidates.